packages/react-magma-dom/src/components/Transition/Transition.tsx

```diff
@@ -0,0 +1,179 @@
+import * as React from 'react';
+import { AnimatePresence, HTMLMotionProps, motion } from 'framer-motion';
+import { MotionVariants } from '../../theme/components/transition';
+
+import { ThemeContext } from '../../theme/ThemeContext';
+import { ThemeInterface } from '../../theme/magma';
+
+/**
+ * @children required
+ */
+export interface BaseTransitionProps extends HTMLMotionProps<'div'> {
+  /**
+   * If `true`, the element will unmount when `in={false}` and animation is done
+   */
+  unmountOnExit?: boolean;
+  /**
+   * If `true`, the content will animate in
+   */
+  in?: boolean;
+  /**
+   * If `true` the children will have the `fade` transition applied from the theme.
+   * @deafult false
+   */
+  fade?: boolean;
+  /**
+   * Custom styles to apply to the transition component div.
+   */
+  style?: React.CSSProperties;
```

Comment:
Do we need the `style` prop, or do we get it by extending attributes from `div`?